scripts/daily-branch-merge.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+TRAVIS_BRANCH=$1
+echo "TRAVIS_BRANCH IS ${TRAVIS_BRANCH}"
+
+containsElement () {
+    local e match="$1"
+    shift
+    for e; do [[ "$e" == "$match" ]] && return 0; done
+    return 1
+}
+
+hasOneMergeConflictMarkerSet () {
+    local fileName="$1"
+    local mergeConflictMarkers="$(git diff --check ${fileName})"
+    local tokens=($(echo $mergeConflictMarkers | tr "\n" "\n"))
+    local numberOfMatches=$(echo "${tokens[@]}" | tr " " "\n" | grep -c "${fileName}:")
+    if [ $numberOfMatches -eq 3 ]; then
+        return 0
+    fi
+    return 1
+}
+
+hasTwoVersionKeys () {
+    local fileName="$1"
+    # Get the content between the merge conflict markers
+    local conflict=$(sed -n -e '/^<<<<<<< HEAD$/,/^>>>>>>> dev$/ { /^<<<<<<< HEAD$/d; /^>>>>>>> dev$/d; p; }' "${fileName}")
+    # Get the number of versions in the string (it should be 2)
+    local numberOfVersions=$(echo "$conflict" | tr " " "\n" | grep -c "version")
+    # Return 0 if there are 2 version keys in the conflict marker
+    if [ $numberOfVersions -eq 2 ]; then
+        return 0
+    fi
+    return 1
+}
+
+# Try to dermine if only package and package-lock is in conflict
+# Then determine if only the version fields are in conflict
+# If the above are true, it is safe to discard changes from these 2 files and proceed with the merge
```

Comment:
Does it mean that during the merge do `dev`, if a node dependency have been introduce in `master` and then is in conflict with another dependency in `dev`, the dependency from `master` will be drop in favour of the one in `dev`? By the way this situation should not happen too much, it might not be a good idea to introduce (or remove) packages directly into `master` (or modify anything else in the `package.json` in this branch).

Reply:
Only the `version` field is considered in the event of a merge conflict where the `version` in `dev` takes precedence over the `version` in `master` presumably because `dev` in further in the future code-wise compared to `master`.

If there are conflicts beyond the `version` field, automatic merging is stopped and has to be done manually (i.e. new dependencies).

Reply:
Perfect 👍